src/module/server-group/ServerGroup.ts

```diff
@@ -34,4 +39,65 @@ module.config(($stateProvider: ng.ui.IStateProvider) => {
         }]
     }
   });
+  $stateProvider.state("server-group.nodes", {
+    url: "/nodes",
+    templateUrl: "module/server-group/nodes/view/server-group.html"
+  });
+
+  $stateProvider.state("server-group.endpoints", {
+    url: "/endpoints",
+    controller: EndpointsCtrl,
+    controllerAs: "ctrl",
+    templateUrl: "module/server-group/endpoints/view/endpoints.html",
+    resolve: {
+      endpoints: ["endpointService", "serverGroup",
+        (endpointService: EndpointService, serverGroup: IServerGroup) => {
+          return endpointService.getAllClusterEndpoints(serverGroup);
+        }],
+    }
+  });
+
+  $stateProvider.state("edit-endpoint-config", {
+    parent: "root",
+    url: "/server-groups/:serverGroup/endpoints/:endpointType/:endpointName/config",
+    controller: EndpointConfigCtrl,
+    controllerAs: "ctrl",
+    templateUrl: "module/server-group/endpoints/config/view/endpoint-config.html",
+    resolve: {
+      serverGroup: ["$stateParams", "serverGroupService", ($stateParams, serverGroupService) => {
+        return serverGroupService.getServerGroupMapWithMembers($stateParams.serverGroup);
+      }],
+      endpoint: ["$stateParams", "endpointService", "serverGroup", ($stateParams, endpointService, serverGroup) => {
+        return endpointService.getEndpoint(serverGroup, $stateParams.endpointType, $stateParams.endpointName);
+      }],
+      endpointMeta: ["$stateParams", "endpointService", "serverGroup", ($stateParams, endpointService, serverGroup) => {
+        return endpointService.getConfigurationMeta(serverGroup.profile,  $stateParams.endpointType,  $stateParams.endpointType);
+      }],
+      endpointName: ["$stateParams", ($stateParams) => $stateParams.endpointName],
+      endpointType: ["$stateParams", ($stateParams) => $stateParams.endpointType]
+    }
+  });
+
+  $stateProvider.state("new-endpoint-config", {
+    parent: "root",
+    url: "/server-groups/:serverGroup/endpoints/:endpointType/:endpointName/newEndpoint",
+    controller: EndpointConfigCtrl,
+    controllerAs: "ctrl",
+    templateUrl: "module/server-group/endpoints/config/view/endpoint-config.html",
+    resolve: {
+      serverGroup: ["$stateParams", "serverGroupService", ($stateParams, serverGroupService) => {
+        return serverGroupService.getServerGroupMapWithMembers($stateParams.serverGroup);
+      }],
+      endpoint: ["$stateParams", ($stateParams) => {
+        let newEndpoint:IEndpoint = EndpointService.parseEndpoint([].concat($stateParams.endpointType).concat($stateParams.endpointName), {});
+        newEndpoint.createdFromUI();
```

Comment:
Yes, the problem is that I don't want any of this logic to leak into configuration section. I agree with your suggestion above!